features/mbedtls/targets/TARGET_NUVOTON/TARGET_NUC472/aes/aes_alt.c

```diff
@@ -159,166 +136,145 @@ void mbedtls_aes_free( mbedtls_aes_context *ctx )
 //    SYS_LockReg();
 
 //    NVIC_DisableIRQ(CRPT_IRQn);
-//    AES_DISABLE_INT();       	
-		channel_free(ctx->channel);
+//    AES_DISABLE_INT();
+    channel_free(ctx->channel);
     mbedtls_zeroize( ctx, sizeof( mbedtls_aes_context ) );
 }
 
 /*
  * AES key schedule (encryption)
  */
-#if defined(MBEDTLS_AES_SETKEY_ENC_ALT)
 int mbedtls_aes_setkey_enc( mbedtls_aes_context *ctx, const unsigned char *key,
-                    unsigned int keybits )
+                            unsigned int keybits )
 {
     unsigned int i;
-	
-	mbedtls_trace("=== %s keybits[%d]\r\n", __FUNCTION__, keybits);
-	dumpHex(key,keybits/8);
-
-    switch( keybits )
-    {
-        case 128: 
-       	    ctx->keySize = AES_KEY_SIZE_128;
-            break;
-        case 192:  
-        	ctx->keySize = AES_KEY_SIZE_192;
-            break;
-        case 256:  
-            ctx->keySize = AES_KEY_SIZE_256;
-            break;
-        default : return( MBEDTLS_ERR_AES_INVALID_KEY_LENGTH );
+
+
+    switch( keybits ) {
+    case 128:
+        ctx->keySize = AES_KEY_SIZE_128;
+        break;
+    case 192:
+        ctx->keySize = AES_KEY_SIZE_192;
+        break;
+    case 256:
+        ctx->keySize = AES_KEY_SIZE_256;
+        break;
+    default :
+        return( MBEDTLS_ERR_AES_INVALID_KEY_LENGTH );
     }
 
 
 
-	// key swap
-		for( i = 0; i < ( keybits >> 5 ); i++ )
-		{
-						ctx->buf[i] = (*(key+i*4) << 24) |
-													(*(key+1+i*4) << 16) |
-													(*(key+2+i*4) << 8) |
-													(*(key+3+i*4) );
-		}
+    // key swap
+    for( i = 0; i < ( keybits >> 5 ); i++ ) {
+        ctx->buf[i] = (*(key+i*4) << 24) |
+                      (*(key+1+i*4) << 16) |
+                      (*(key+2+i*4) << 8) |
+                      (*(key+3+i*4) );
+    }
     AES_SetKey(ctx->channel, ctx->buf, ctx->keySize);
 
 
     return( 0 );
 }
-#endif /* MBEDTLS_AES_SETKEY_ENC_ALT */
 
 /*
  * AES key schedule (decryption)
  */
-#if defined(MBEDTLS_AES_SETKEY_DEC_ALT)
 int mbedtls_aes_setkey_dec( mbedtls_aes_context *ctx, const unsigned char *key,
-                    unsigned int keybits )
+                            unsigned int keybits )
 {
     int ret;
-	
-	  mbedtls_trace("=== %s keybits[%d]\r\n", __FUNCTION__, keybits);
-	  dumpHex((uint8_t *)key,keybits/8);
-	
+
+
     /* Also checks keybits */
     if( ( ret = mbedtls_aes_setkey_enc( ctx, key, keybits ) ) != 0 )
-        goto exit;    
+        goto exit;
 
 exit:
 
     return( ret );
 }
-#endif /* MBEDTLS_AES_SETKEY_DEC_ALT */
 
 
 static void __nvt_aes_crypt( mbedtls_aes_context *ctx,
-                          const unsigned char input[16],
-                          unsigned char output[16], int dataSize)
+                             const unsigned char input[16],
```

Comment:
I missed this in the previous Review.
This function also seems not thread safe. `au8OutputData` and `au8InputData` are defined out of this function, as stsatic variables, which could be overwritten every cryptographic call